cmd/disk-cache.go

```diff
@@ -711,14 +710,12 @@ func (c *cacheObjects) gc(ctx context.Context, doneCh chan struct{}) {
 				if dcache.gcCount() == 0 {
 					continue
 				}
-				gcLock.Lock()
 				wg.Add(1)
-				go func(d *diskCache, l *sync.Mutex) {
+				go func(d *diskCache) {
```

Comment:
Wait. I thought this would only start if we reached the high watermark?

Reply:
that is correct @klauspost - line 710 checks for this. When high watermark is reached, the gcCounter gets incremented for each cache access until gc is started on that drive.